sql/LabelCenterline.sql

```diff
@@ -0,0 +1,71 @@
+/******************************************************************************
+### LabelCenterline ###
+
+Given a polygon or multipolygon, calculates a linestring suitable for a label.
+
+__Parameters:__
+
+- `geometry` inGeometry - A polygon or multipolygon.
+
+__Returns:__ `geometry(multiline)`
+******************************************************************************/
+CREATE OR REPLACE FUNCTION CountDisconnectedEndpoints(polyline geometry, testline geometry)
+    RETURNS integer AS $$
+    BEGIN
+        RETURN ST_NPoints(ST_RemoveRepeatedPoints(ST_Points(polyline)))
+                    - ST_NPoints(ST_RemoveRepeatedPoints(ST_Points(ST_Difference(polyline, testline))))
+                    - ST_NPoints(testline) + 2;
+    END
+    $$ LANGUAGE plpgsql;
+CREATE OR REPLACE FUNCTION TrimmedCenterline(inPolyline geometry)
+    RETURNS geometry AS $$
+    DECLARE outPolyline geometry;
+    BEGIN
+        WITH tbla AS (
```

Comment:
More meaningful CTE names would be good.

Reply:
Good point!